cassandra_backup_service.py

```diff
@@ -1320,6 +1326,32 @@ def save_manifest(self, keyspace, columnfamily, manifest):
 
         return manifest_file_path
 
+    def manifest_data_retention_pruning(self, data):
```

Comment:
What is the format of data? Isn't this something like:  `data['incremental'][filename]['created'] = time` per line 1402?.. If so how  does `for inc_ts in data['incremental']:` ... `int(inc_ts)` work?

Reply:
The data here looks like the following:

```python

    def manifest_data_retention_pruning(self, data):
        """
        Prune provided data to only values within the defined self.retention_days time range.

        :param dict data: manifest data.
        """
        print data
        if 'full' in data:
```

```
{
    u'column_family': u'compaction_history',
    u'keyspace': u'system',
    u'updated': u'2021-01-05 00:58:59',
    u'full': {
        u'1609808336967': {
            u'lb-108-big-Filter.db': {u'created': u'2021-01-05 00:58:58'},
            u'lb-108-big-Summary.db': {u'created': u'2021-01-05 00:58:58'},
            u'lb-108-big-Digest.adler32': {u'created': u'2021-01-05 00:58:58'},
            u'lb-107-big-Index.db': {u'created': u'2021-01-04 23:05:42'},
[...]
```

The `int` casting works on the key which is the timestamp with microseconds.

Reply:
@bogdan3d what does the incremental data look like?